changelog/580.feature

```diff
@@ -0,0 +1 @@
+make pytest.fail/skip/xfail instances of BaseException
```

Comment:
Also if we will release a new major version, should we add this in a "backward incompatibility" section perhaps?